src/BlockTypes/ProductGallery.php

```diff
@@ -90,12 +98,14 @@ function( $carry, $item ) {
 			</div>
 			</div>
 			<div class="wc-block-product-gallery-dialog__body">
+				<h2>{{product_title}}</h2>
```

Comment:
The Product Title should be added in the template part:
<img width="1322" alt="Product_Gallery_‹_Template_part_‹_Editor_‹_newproductgallery_—_WordPress" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/218afe89-56b9-4aaa-9619-d416ddaf2b4b">